independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/ClassPathResourceIndex.java

```diff
@@ -117,12 +117,12 @@ public List<ClassPathElement> getClassPathElements(String resource) {
         }
 
         List<ClassPathElement> classPathElements = new ArrayList<>(candidates.length);
-        for (ClassPathElement candidate : candidates) {
-            if (candidate.getProvidedResources().contains(resource)) {
-                classPathElements.add(candidate);
+        for (int i = 0; i < candidates.length; i++) {
+            if (candidates[i].getProvidedResources().contains(resource)) {
+                classPathElements.add(candidates[i]);
             }
         }
-        return Collections.unmodifiableList(classPathElements);
+        return classPathElements;
```

Comment:
I think given it's a hot path and not really user-consumable APIs, we could get rid of this safety.